patientsearch/api.py

```diff
@@ -241,15 +244,38 @@ def resource_bundle(resource_type):
 
     """
     token = validate_auth()
+    # Check for the store's configurations
+    active_patient_flag = current_app.config.get("ACTIVE_PATIENT_FLAG")
+    params = dict(deepcopy(request.args))  # Necessary on ImmutableMultiDict
+
+    # Override if the search is specifically for inactive objects, only occurs
+    # when working on reactivating a patient
+    inactive_search_param = request.args.get("inactive_search")
+    is_inactive_search = False
+    if inactive_search_param:
+        is_inactive_search = inactive_search_param.lower() in {"true", "1"}
+    if is_inactive_search:
+        active_patient_flag = False
+    # need to remove this param, name/key error is raised when passed to HAPI_request
+    if inactive_search_param is not None:
+        del params["inactive_search"]
+
+    # If the resource is a patient, proceed with the active GET, if requested
     try:
-        return jsonify(
-            HAPI_request(
-                token=token,
-                method="GET",
-                resource_type=resource_type,
-                params=request.args,
-            )
+        if (
+            resource_type == "Patient"
```

Comment:
i don't see the need to restrict this to `Patient` - might someday use for `Practitioner` and a host of others.

Reply:
ah, right.  until we add the `active` flag to the other resources, it will exclude any that don't have it set.  I take it back, I get exactly why you need to confirm it's a `resource_type == "Patient"` in determining if `active` should be part of the parameters.

still, these are general purpose methods, can't assume it's only used for patients.